src/app/components/upload-card/UploadCardRenderer.tsx

```diff
@@ -129,7 +138,19 @@ export function UploadCardRenderer({
       bottom={
         <>
           {fileItem.originalFile.type.startsWith('image') && (
-            <ImagePreview fileItem={fileItem} onSpoiler={handleSpoiler} />
+            <MediaPreview
+              fileItem={fileItem}
+              onSpoiler={handleSpoiler}
+              renderPreviewElement={(p) => <img {...p} alt={p.alt} />}
```

Comment:
instead of taking the props from `MediaPreview` we can pass the `fileItem` to new `PreviewImage` component, and render it as `children` instead of `renderPreviewElement`. Doing so will decouple the children (preview image/video) from *holder* UI `MediaPreview`.